### DISCUSSION TITLE
Syntax Error: Expected Name, found "$"

### DISCUSSION THREAD

**Discussion by janat08** [gatsbyjs/gatsby#32179] [Help] [answered]
I'm trying to not use the production/remote database during development.
```
export const query = graphql`
  query ($db_type: Name) {
    $db_type {
      allMovie{
        data {
          title_long
          slug
        }
      }
    }
  }
`
```

**Comment by LekoArts** [comment, marked as answer]
You can't use a variable in this place there. And you probably also shouldn't make the distinction there. Keep the same root query names and switch the source in your source plugin with e.g. an environment variable
